server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -277,7 +278,8 @@ private Long parseNullValue(DateFieldType fieldType) {
             try {
                 return fieldType.parse(nullValue.getValue());
             } catch (Exception e) {
-                DEPRECATION_LOGGER.deprecate("date_mapper_null_field", "Error parsing [" + nullValue.getValue()
+                DEPRECATION_LOGGER.deprecate(DeprecationCategory.MAPPINGS, "date_mapper_null_field",
+                    "Error parsing [" + nullValue.getValue()
```

Comment:
extreme, probably-shouldn't-even-write-this-comment nitpick: putting the line break after `field ["` might be slightly more visually appealing

(I hope these tiny nitpicks give you confidence in the thoroughness of my review, rather than just annoying you.)